2017/1/day1.go

```diff
@@ -0,0 +1,68 @@
+package main
+
+import (
+	"fmt"
+	"io/ioutil"
+	"os"
+	"strconv"
+)
+
+func intFromRune(r rune) (int64, error) {
+	s_r := string(r)
+	v, err := strconv.ParseInt(s_r, 10, 64)
+	if err != nil {
+		return 0, err
+	}
+	return v, nil
+}
+
+func nextPart2(i int, input string) int {
+	// Last char is a line feed
+	real_len := len(input) - 1
+	// The length of the input will always be divisible by 2
+	half := real_len / 2
+	return (i + half) % real_len
+}
+
+func nextPart1(i int, input string) int {
+	// Last char is a line feed
+	if i == len(input)-2 {
+		return 0
+	} else {
+		return i + 1
+	}
+}
+
+func main() {
+	dat, err := ioutil.ReadFile("input.txt")
+	if err != nil {
+		fmt.Printf("Couldn't read file: %v\n", err)
+		os.Exit(1)
+	}
+	input := string(dat)
+
+	var sum int64
+	for i, c := range input {
```

Comment:
i've realized that if i actually iterate over the bytes i can:
- avoid explicitly skipping the linefeed everywhere
- avoid converting from rune to int

Reply:
👍